analytics/build/build.go

```diff
@@ -91,3 +100,32 @@ func (ea enabledAnalytics) LogNotificationEventObject(ne *analytics.Notification
 		}
 	}
 }
+
+func evaluateActivities(rw *openrtb_ext.RequestWrapper, ac privacy.ActivityControl, ap config.AccountPrivacy, componentName string) (bool, *openrtb_ext.RequestWrapper) {
```

Comment:
Nitpick: I re-worked this function so that if statements aren't nested for improved readability, this is personally how I prefer it, but just wanted to suggest it

```
func evaluateActivities(rw *openrtb_ext.RequestWrapper, ac privacy.ActivityControl, ap config.AccountPrivacy, componentName string) (bool, *openrtb_ext.RequestWrapper) {
    component := privacy.Component{Type: privacy.ComponentTypeAnalytics, Name: componentName}
    if !ac.Allow(privacy.ActivityReportAnalytics, component, privacy.ActivityRequest{}) {
        return false, nil
    }

    blockUserFPD := !ac.Allow(privacy.ActivityTransmitUserFPD, component, privacy.ActivityRequest{})
    blockPreciseGeo := !ac.Allow(privacy.ActivityTransmitPreciseGeo, component, privacy.ActivityRequest{})

    if !blockUserFPD && !blockPreciseGeo {
        return true, nil
    }

    cloneReq := ortb.CloneBidderReq(rw.BidRequest)

    if blockUserFPD {
        privacy.ScrubUserFPD(cloneReq)
    }
    if blockPreciseGeo {
        ipConf := privacy.IPConf{IPV6: ap.IPv6Config, IPV4: ap.IPv4Config}
        privacy.ScrubGeoAndDeviceIP(cloneReq, ipConf)
    }

    cloneReq.RebuildRequest()
    return true, cloneReq
}
```

Reply:
Agree, modified. Good point Alex!